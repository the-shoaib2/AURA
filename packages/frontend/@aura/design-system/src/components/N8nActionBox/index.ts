import auraActionBox from './ActionBox.vue';

export default auraActionBox;
