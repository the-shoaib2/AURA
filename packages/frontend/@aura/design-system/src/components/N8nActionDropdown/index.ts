import auraActionDropdown from './ActionDropdown.vue';
export default auraActionDropdown;
