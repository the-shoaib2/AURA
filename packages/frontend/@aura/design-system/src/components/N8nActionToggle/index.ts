import auraActionToggle from './ActionToggle.vue';

export default auraActionToggle;
