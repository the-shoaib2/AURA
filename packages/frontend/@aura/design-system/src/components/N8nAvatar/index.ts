import auraAvatar from './Avatar.vue';

export default auraAvatar;
