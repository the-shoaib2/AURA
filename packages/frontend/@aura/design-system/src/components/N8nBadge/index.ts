import auraBadge from './Badge.vue';

export default auraBadge;
