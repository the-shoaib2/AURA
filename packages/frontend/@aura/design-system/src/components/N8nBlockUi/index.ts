import auraBlockUi from './BlockUi.vue';

export default auraBlockUi;
