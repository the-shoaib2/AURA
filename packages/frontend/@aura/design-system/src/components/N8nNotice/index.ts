import auraNotice from './Notice.vue';

export default auraNotice;
