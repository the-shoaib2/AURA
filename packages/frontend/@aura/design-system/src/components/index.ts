export { default as auraActionBox } from './auraActionBox';
export { default as auraActionDropdown } from './auraActionDropdown';
export { default as auraActionToggle } from './auraActionToggle';
export { default as auraAlert } from './auraAlert';
export { default as auraAvatar } from './auraAvatar';
export { default as auraBadge } from './auraBadge';
export { default as auraBlockUi } from './auraBlockUi';
export { default as auraButton } from './auraButton';
export { default as auraCallout } from './auraCallout';
export { default as auraCard } from './auraCard';
export { default as auraCheckbox } from './auraCheckbox';
export { default as auraCircleLoader } from './auraCircleLoader';
export { default as auraColorPicker } from './auraColorPicker';
export { default as auraDatatable } from './auraDatatable';
export { default as auraExternalLink } from './auraExternalLink';
export { default as auraFormBox } from './auraFormBox';
export { default as auraFormInputs } from './auraFormInputs';
export { default as auraFormInput } from './auraFormInput';
export { default as auraHeading } from './auraHeading';
export { default as auraIcon } from './auraIcon';
export { default as auraIconButton } from './auraIconButton';
export { default as auraInfoAccordion } from './auraInfoAccordion';
export { default as auraInfoTip } from './auraInfoTip';
export { default as auraInput } from './auraInput';
export { default as auraInputLabel } from './auraInputLabel';
export { default as auraInputNumber } from './auraInputNumber';
export { default as auraLink } from './auraLink';
export { default as auraLoading } from './auraLoading';
export { default as auraMarkdown } from './auraMarkdown';
export { default as auraMenu } from './auraMenu';
export { default as auraMenuItem } from './auraMenuItem';
export { default as auraNavigationDropdown } from './auraNavigationDropdown';
export { default as auraNodeCreatorNode } from './auraNodeCreatorNode';
export { default as auraNodeIcon } from './auraNodeIcon';
export { default as auraNotice } from './auraNotice';
export { default as auraOption } from './auraOption';
export { default as auraSelectableList } from './auraSelectableList';
export { default as auraPopover } from './auraPopover';
export { default as auraPulse } from './auraPulse';
export { default as auraRadioButtons } from './auraRadioButtons';
export { default as auraRoute } from './auraRoute';
export { default as auraRecycleScroller } from './auraRecycleScroller';
export { default as auraResizeWrapper } from './auraResizeWrapper';
export { default as auraSelect } from './auraSelect';
export { default as auraSpinner } from './auraSpinner';
export { default as auraSticky } from './auraSticky';
export { default as auraResizeableSticky } from './auraResizeableSticky';
export { default as auraTabs } from './auraTabs';
export { default as auraTag } from './auraTag';
export { default as auraTags } from './auraTags';
export { default as auraText } from './auraText';
export { default as auraTooltip } from './auraTooltip';
export { default as auraTree } from './auraTree';
export { default as auraUserStack } from './auraUserStack';
export { default as auraUserInfo } from './auraUserInfo';
export { default as auraUserSelect } from './auraUserSelect';
export { default as auraUsersList } from './auraUsersList';
export { default as auraResizeObserver } from './ResizeObserver';
export { auraKeyboardShortcut } from './auraKeyboardShortcut';
export { default as auraIconPicker } from './auraIconPicker';
export { default as auraBreadcrumbs } from './auraBreadcrumbs';
export { default as auraTableBase } from './TableBase';
export { default as auraDataTableServer } from './auraDataTableServer';
export { default as auraTableHeaderControlsButton } from './TableHeaderControlsButton';
export { default as auraInlineTextEdit } from './auraInlineTextEdit';
export { default as auraScrollArea } from './auraScrollArea';
