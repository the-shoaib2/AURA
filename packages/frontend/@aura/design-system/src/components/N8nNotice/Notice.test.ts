import { render } from '@testing-library/vue';

import { auraText } from '@aura/design-system/components';
import { auraHtml } from '@aura/design-system/directives';

import auraNotice from './Notice.vue';

describe('components', () => {
	describe('auraNotice', () => {
		it('should render correctly', () => {
			const wrapper = render(auraNotice, {
				props: {
					id: 'notice',
				},
				slots: {
					default: 'This is a notice.',
				},
				global: {
					stubs: ['aura-text'],
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});

		describe('props', () => {
			describe('content', () => {
				it('should render correctly with content prop', () => {
					const wrapper = render(auraNotice, {
						props: {
							id: 'notice',
							content: 'This is a notice.',
						},
						global: {
							stubs: ['aura-text'],
						},
					});
					expect(wrapper.html()).toMatchSnapshot();
				});

				it('should render HTML', () => {
					const wrapper = render(auraNotice, {
						props: {
							id: 'notice',
							content: '<strong>Hello world!</strong> This is a notice.',
						},
						global: {
							directives: {
								auraHtml,
							},
							components: {
								'aura-text': auraText,
							},
						},
					});

					expect(wrapper.container.querySelectorAll('strong')).toHaveLength(1);
					expect(wrapper.html()).toMatchSnapshot();
				});

				it('should sanitize rendered HTML', () => {
					const wrapper = render(auraNotice, {
						props: {
							id: 'notice',
							content: '<script>alert(1);</script> This is a notice.',
						},
						global: {
							stubs: ['aura-text'],
						},
					});

					expect(wrapper.container.querySelector('script')).not.toBeTruthy();
					expect(wrapper.html()).toMatchSnapshot();
				});
			});
		});
	});
});
