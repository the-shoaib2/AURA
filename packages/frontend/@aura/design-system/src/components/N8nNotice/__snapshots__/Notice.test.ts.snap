// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components > auraNotice > props > content > should render HTML 1`] = `
"<div id="notice" class="notice notice warning" role="alert">
  <div class="notice-content"><span class="aura-text compact size-small regular"><span id="notice-content" class="truncated" role="region"><strong>Hello world!</strong> This is a notice.</span></span></div>
</div>"
`;

exports[`components > auraNotice > props > content > should render correctly with content prop 1`] = `
"<div id="notice" class="notice notice warning" role="alert">
  <div class="notice-content">
    <aura-text-stub bold="false" size="small" compact="true" tag="span"></aura-text-stub>
  </div>
</div>"
`;

exports[`components > auraNotice > props > content > should sanitize rendered HTML 1`] = `
"<div id="notice" class="notice notice warning" role="alert">
  <div class="notice-content">
    <aura-text-stub bold="false" size="small" compact="true" tag="span"></aura-text-stub>
  </div>
</div>"
`;

exports[`components > auraNotice > should render correctly 1`] = `
"<div id="notice" class="notice notice warning" role="alert">
  <div class="notice-content">
    <aura-text-stub bold="false" size="small" compact="true" tag="span"></aura-text-stub>
  </div>
</div>"
`;
